services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/oss/OssUtil.java

```diff
@@ -46,6 +56,152 @@
+                // Safe removal of quotes
+                quotedList.add(term.substring(1, length - 1));
+            } else {
+                unquotedList.add(term);
+            }
+        });
+
+        addTermSuggester(suggesterBuilder, size, maxEdits);
+        suggesterBuilder.text(String.join(" ", unquotedList));
+
+        Suggester suggester = suggesterBuilder.build();
+        return searchBuilder.suggest(suggester).build();
+    }
+
+    public static void addTermSuggester(Suggester.Builder suggesterBuilder, int size, int maxEdits) {
+        suggesterBuilder.suggesters(TERM_NAME_SUGGESTION, s -> s.term(
+                ts -> ts
+                        .field(NAME_FIELD)
+                        .size(size)
+                        .maxEdits(maxEdits)
+                        .suggestMode(SuggestMode.Missing)
+        ));
+        suggesterBuilder.suggesters(TERM_DESCRIPTION_SUGGESTION, s -> s.term(
+                ts -> ts
+                        .field(DESCRIPTION_FIELD)
+                        .size(size)
+                        .maxEdits(maxEdits)
+                        .suggestMode(SuggestMode.Missing
+                        )));
+    }
+
+    public static SuggestionResults convertToSynapseSuggestionResult(Map<String, List<Suggest<DocumentFields>>> suggestions) {
+        ValidateArgument.required(suggestions, "suggestions");
+        SuggestionResults results = new SuggestionResults();
+        Map<String, Set<Option>> map1 = new HashMap<>();
+
+        suggestions.entrySet().stream()
+                .map(Map.Entry::getValue)
+                .filter(suggestList -> !suggestList.isEmpty())
+                .flatMap(List::stream)
+                .filter(suggest -> suggest != null && suggest.term() != null)
+                .forEach(suggest -> {
+                    Set<Option> optionSet = map1.computeIfAbsent(suggest.term().text(), k -> new HashSet<>());
+                    suggest.term().options().stream()
+                            .filter(option -> option != null && StringUtils.isNotEmpty(option.text()))
+                            .map(option -> new Option()
+                                    .setTerm(option.text())
+                                    .setScore(option.score()))
+                            .forEach(optionSet::add);
+                });
+
+
+        List<Suggestion> suggestionList = map1.entrySet()
+                .stream()
+                .map(entry -> { // 2. Transform each entry into a Suggestion object
+                    return new Suggestion()
+                            .setKey(entry.getKey())
+                            .setValues(entry.getValue());
+                })
+                .collect(Collectors.toList());
+
+        return results.setSuggestions(suggestionList);
+
+    }
+
+    public static SuggestionResults eliminateSuggestionWithAccessDenied(SuggestionResults suggestionResults, Map<String, Aggregate> aggregateResponse) {
```

Comment:
can you add a comment explaining what this function does, e.g.:  Filters suggestion results to return only the results which appear in a document the user can see (represented in 'aggregateResponse')?

Might it be better to pass the parameter `filteredBucketMap` instead of `aggregateResponse`?  I.e., this function only needs that map.